src/citrine/resources/design_workflow.py

```diff
@@ -18,10 +20,44 @@ class DesignWorkflowCollection(Collection[DesignWorkflow]):
     _resource = DesignWorkflow
     _api_version = "v2"
 
-    def __init__(self, project_id: UUID, session: Session, branch_id: UUID = None):
+    def __init__(self,
+                 project_id: UUID,
+                 session: Session,
+                 branch_id: Optional[UUID] = None,
+                 branch_root_id: Optional[UUID] = None,
+                 branch_version: Optional[int] = None):
```

Comment:
It doesn't make sense that these would all be positional arguments.  Allowing three is already outside [repo coding style](https://github.com/CitrineInformatics/citrine-python/blob/main/CONTRIBUTING.md#codestyle).  If you're migrating, we should probably do something like:
```suggestion
    def __init__(self,
                 project_id: UUID,
                 session: Session,
                 positional_branch_id: Optional[UUID] = None,
                 *,
                 branch_id: Optional[UUID] = None,
                 branch_root_id: Optional[UUID] = None,
                 branch_version: Optional[int] = None):
```
and throw a deprecation warning when the `positional_branch_id` argument is not None.  Alternatively, if you intend to support positional arguments for `branch_root_id` & `branch_version` and `branch_id` and `branch_root_id` are functionally interchangeable during the migration,
```suggestion
    def __init__(self,
                 project_id: UUID,
                 session: Session,
                 branch_root_id: Optional[UUID] = None,
                 branch_version: Optional[int] = None):
                 *,
                 branch_id: Optional[UUID] = None,
```
and deprecation is issued when branch_id is not None.

Reply:
I'm not sure if any of our other collection constructors have keyword-only parameters. But maybe they should?

In any case, sure, I'll make the new args kw-only. But since we're deprecating `branch_id`, I'm going to leave it as is.